trie/sparse_merkle_tree.py

```diff
@@ -0,0 +1,108 @@
+from eth_hash.auto import (
+    keccak,
+)
+
+from trie.constants import (
+    EMPTY_LEAF_NODE_HASH,
+    EMPTY_NODE_HASHES,
+)
+from trie.validation import (
+    validate_is_bytes,
+    validate_length,
+)
+
+
+# sanity check
+assert EMPTY_LEAF_NODE_HASH == keccak(b'')
+
+
+class SparseMerkleTree:
+    def __init__(self, db):
+        self.db = db
+        # Initialize an empty tree with one branch
+        self.root_hash = keccak(EMPTY_NODE_HASHES[0] + EMPTY_NODE_HASHES[0])
+        self.db[self.root_hash] = EMPTY_NODE_HASHES[0] + EMPTY_NODE_HASHES[0]
+        for i in range(159):
+            self.db[EMPTY_NODE_HASHES[i]] = EMPTY_NODE_HASHES[i+1] + EMPTY_NODE_HASHES[i+1]
+        self.db[EMPTY_LEAF_NODE_HASH] = b''
+
+    def get(self, key):
+        validate_is_bytes(key)
+        validate_length(key, 20)
+
+        target_bit = 1 << 159
+        path = int.from_bytes(key, byteorder='big')
+        node_hash = self.root_hash
+        for i in range(160):
+            if path & target_bit:
+                node_hash = self.db[node_hash][32:]
+            else:
+                node_hash = self.db[node_hash][:32]
+            target_bit >>= 1
+
+        if self.db[node_hash] is b'':
+            return None
```

Comment:
Reason for returning `None` instead of raising a `KeyError`?  We've found that return value checking is way easier to forget than exception handling.

Reply:
No particular reason. Just that in hexary trie, `BLANK_NODE` is returned for non-existent key. Raising an exception does make more sense.